src/components/BuyTickets/index.tsx

```diff
@@ -0,0 +1,54 @@
+import style from "./style.module.scss";
+import { ScreeningProps } from "@/types/screeningTypes";
+import React, { useState } from "react";
+import { useRouter } from "next/router";
+import { NumberContextProvider } from "@/util/NumberContext";
+import { MovieProps } from "@/types/movieTypes";
+
+export default function BuyTickets({
+  screenings,
+  selectedSeatIds,
+  id,
+  displayDate,
+  movieData,
+}: {
+  screenings: ScreeningProps;
+  selectedSeatIds: number[];
+  id: string;
+  displayDate: string;
+  movieData: MovieProps;
+}) {
+  const [seats, setSeats] = useState(selectedSeatIds);
+
+  let dateAndTime = new Date(parseInt(displayDate));
+  console.log(dateAndTime);
+  const ISOdate = dateAndTime.toISOString();
+  const offsetFormattedDate = ISOdate.replace("T", " ").slice(0, 16);
+  const Link = `/movies/${id}/booking/${offsetFormattedDate}/payment`;
+  const router = useRouter();
+  function handleClick() {
+    setSeats(selectedSeatIds);
+    router.push(Link);
+  }
+  return (
+    <div className={style.container}>
+      <NumberContextProvider
+        numberArray={selectedSeatIds}
+      ></NumberContextProvider>
```

Comment:
The way this provider is used, it doesn't actually share the context with anyone.

A provider is a component that creates a context value, and _provides_ it to all of it's _children_ (and their children, and their children etc). Here, the provider has no children, so it effectively does nothing.